### ISSUE TITLE
Items are never removed from map in EntityProvider

### ISSUE THREAD

**Issue by SSS135** [scellecs/morpeh#214]
Hi, I've noticed that in EntityProvider.cs `--item.refCounter` affects only the local item struct, not the one stored in `map`.

Current code:
```
if (map.TryGetValue(instanceId, out var item)) {
    if (--item.refCounter <= 0) {
        map.Remove(instanceId, out _);
    }
}
```

Fixed code:
```
if (map.TryGetValue(instanceId, out var item))
{
    item.refCounter--;
    if (item.refCounter <= 0) {
        map.Remove(instanceId, out _);
    } else {
        map.Set(instanceId, item, out _);
    }
}
```

**Comment by olegmrzv** [comment]
Will be fixed soon.

**Comment by olegmrzv** [comment]
Fixed in [2023.1.1-rc11](https://github.com/scellecs/morpeh/tree/2023.1.1-rc11)
2023.1.1 will be released in March when the Stash.Clone API is finished
